### PR TITLE
feat: add script.name attribute.

### PR DESCRIPTION
**Pull request by i4ki**
## What this PR does / why we need it:

Add an optional `script.name` attribute. This is purely a cosmetic change.
Now the commands `terramate script info`, `terramate script list` and `terramate script tree` show the name if available. Example below:

Screenshot 2024-02-22 at 19 17 28
Screenshot 2024-02-22 at 19 17 51
Screenshot 2024-02-22 at 19 18 09

## Which issue(s) this PR fixes:

none

## Does this PR introduce a user-facing change?
```
yes, a new script.name attr added.
```

### CODE REVIEW

File: hcl/hcl_script.go [hunk 1]
```diff
@@ -65,8 +75,8 @@ func NewScriptCommands(attr ast.Attribute) *Commands {
 	return &cmds
 }
 
-// Name returns the formatted script name
-func (sc *Script) Name() string {
+// AcessorName returns the name traversal for acessing the script.
+func (sc *Script) AcessorName() string {
```
Comment (wmalik) [inline hcl/hcl_script.go:79]:
```suggestion
// AccessorName returns the name traversal for accessing the script.
func (sc *Script) AccessorName() string {
```

File: config/script.go [hunk 2]
```diff
@@ -47,8 +48,12 @@ type ScriptJob struct {
 type Script struct {
 	Range       info.Range
 	Labels      []string
+	Name        string
 	Description string
 	Jobs        []ScriptJob
+
+	// List of warnings to be rendered at a later point.
+	Warnings []string
```
Comment (wmalik) [inline config/script.go:56]: this seems to be unused

**Review by wmalik — COMMENTED** [review]
added some questions

File: cmd/terramate/cli/script_run.go [hunk 3]
```diff
@@ -179,6 +179,7 @@ func (c *cli) prepareScriptCloudDeploymentSync(runStacks []runContext) {
 // for example:
 // /somestack (script:0 job:0.0)> echo hello
 func printScriptCommand(w io.Writer, run runContext) {
+	// TODO(i4k): print script.name?
```
[the remark this answers is not in the document]
Comment (i4ki) [reply cmd/terramate/cli/script_run.go:182]: After some tests, I dont feel like it fits here... it pollutes the output and if we shorten it too much then the name does not make sense anymore.

**Review by snakster — APPROVED** [review]
Looks good! 👍

**Pull request merged**